### DISCUSSION TITLE
Need help with setting up docker environment and start/stop the server from Python Script

### DISCUSSION THREAD

**Discussion by lav20013** [mockoon/mockoon#966] [Help]
I was able to docker container with below entry in docker file
FROM node:14-alpine

RUN npm install -g @mockoon/cli@2.3.0
COPY Mockoon/FullCircle.json ./mockoon-fullcircle.json
COPY Mockoon/JobOrders/ ./JobOrders/
RUN adduser --shell /bin/sh --disabled-password --xxxx "" mockoon
RUN chown -R mockoon ./mockoon-fullcircle.json
USER mockoon

EXPOSE 3001

ENTRYPOINT ["mockoon-cli", "start", "--hostname", "0.0.0.0", "--daemon-off", "--data", "mockoon-fullcircle.json", "--container"]

RUN -d -p <host_port>:3001 mockoon-mock1 --log-transaction

This last entry in docker file is giving error when building docker image on jenkins pipeline
illegal option -d

Also wanted to know how to start and stop the server from Python script ,how to launch url and use mocks from Python,
Any help will be appreciated.

**Comment by 255kb** [comment]
Isn't `-d` in `RUN -d ...` a `docker compose up` flag only? 

Concnerning your other questions, your Python script would have to run the Docker commands, like `docker stop` and `docker run`. 
I am not that good with Docker, but I don't think I would put the `RUN` directly in the Dockerfile. But instead I would run the image after the build with `docker run -p <host_port>:3001 mockoon-mock1 --log-transaction`.

To use your mock in Python, just do a request to the exposed port like you would do with any other running application.

**Reply by lav20013** [reply]
Thank you able to make it work ,how do i check after docker run -p <host_port>:3001 mockoon-mock1 --log-transaction.?
Trying to understand next steps on how to check if mock is started up and running.

**Reply by 255kb** [reply]
As soon as your mock is running in Docker and the port exposed, in your example 3001, you can make HTTP calls to the running mock. 
For example, if you declared a GET /test endpoint, you would be able to use curl: `curl http://localhost:3001/test`. You can also use other tools like Insomnia, Postman or Hoppscotch to make API calls.

**Reply by lav20013** [reply]
Thank you this is running on docker container on some other machine, how can i test in my local?

**Reply by 255kb** [reply]
I am not sure to understand what you want to do. 
But if you want to test your running mock in Docker locally you can run your Docker image locally by using the `docker run` command. Then the same rules apply. You will be able to call your mock endpoints on the exposed port on localhost.
